docs/architecture/adr-016-social-upgrades.md

```diff
@@ -0,0 +1,170 @@
+# ADR 016: Removing Token Voting for Upgrades
+
+## Status
+
+Proposed
+
+## Changelog
+
+- 2023/03/15: initial draft
+
+## Context
+
+Standard cosmos-sdk based chains are sovereign and have the ability to hardfork, but the current implementation makes heavy use of onchain token voting. This bypasses social consensus, one of the core principles of Celestia. After a proposal is submitted and crosses some threshold of agreement, the halt height is set in the state machine for all nodes, and validators are expected to switch binaries after the state machine halts the node. The ultimate security, ruleset, and result of that ruleset rests solely on the shoulders of node operators, not validators or token holders. Therefore we are seeking a solution that engraves social consensus into the implementation itself.
+
+The most pertinent issue at the moment is that we are launching mainnet soon, and don't have a fully functional upgrade mechanism that empowers social consensus in place. We need to first decide how to remove token voting in a way that supports our future efforts to change the upgrade mechanism. The latter discussion on how to actually implement upgrades that fit all of our desired properties is out of scope for this ADR, and will be discussed separately.
+
+It should also be noted that this ADR is not discussion parameter changes, as limiting token voting on those is also a separate discussion.
+
+## Alternative Approaches
+
+### Option 1: Temporarily use the current token voting mechanism
+
+Given that we don't have a lot of time until mainnet, we could leave the current implementation in place which gives us the option to use it if needed. Ideally, we would work on the longer term upgrade mechanism that respects social consensus and finish it before the first upgrade. While this option does allow for maximum flexibility, it is also very risky because if the mechanism is ever needed or used, then it could set a precedent for future upgrades.
+
+### Option 2: Predetermined halt height and removal of token voting for upgrades (aka "difficulty bomb")
```

Comment:
[question] is there an Option 3 that removes token voting but doesn't introduce the difficulty bomb?

Reasons why a difficulty bomb may be unnecessary:
1. There isn't a need to introduce an upgrade mechanism prior to the next (currently unplanned) upgrade. Given the next upgrade isn't planned, we don't have a block height for it. Since no block height exists, we can't implement a difficulty bomb that coincides with when such an upgrade were to take place. Implementing a difficulty bomb with an estimated block height may result in several subsequent patches to delay the bomb.
2. The protocol may not need the difficulty bomb as an incentive to implement an upgrade mechanism. If the next upgrade brings improvements to the protocol, protocol participants may be sufficiently incentivized to introduce an upgrade mechanism to benefit from the proposed changes.

Reply:
we could yeah, but here I didn't want to specify all of the different combinations, as this was just a quick ADR to cover something that was discussed synchronously. 

> There isn't a need to introduce an upgrade mechanism prior to the next (currently unplanned) upgrade. Given the next upgrade isn't planned, we don't have a block height for it. Since no block height exists, we can't implement a difficulty bomb that coincides with when such an upgrade were to take place. Implementing a difficulty bomb with an estimated block height may result in several subsequent patches to delay the bomb.

Adding the difficulty bomb is inherently planning for an upgrade within some period of time. It's explicitly telling the validators that they have to at least make one change before that determined height.

> The protocol may not need the difficulty bomb as an incentive to implement an upgrade mechanism. If the next upgrade brings improvements to the protocol, protocol participants may be sufficiently incentivized to introduce an upgrade mechanism to benefit from the proposed changes.

The attack scenario is that the validators choose not to upgrade but social consensus does. This is just an explicit form of that. Participants could want to upgrade, but that does not mean validators will want to. I can add a line making this more explicit though!

Reply:
I'll add the third option though